apps/clock/js/alarm.js

```diff
@@ -156,6 +156,48 @@ define(function(require, exports, module) {
       // Only update the application if this alarm was actually saved
       // (i.e. it has an ID).
       if (this.id) {
+        var self = this,
```

Comment:
And although it's in Clock app, the reason of avoiding refering `self` regards JavaScript practice, is that `self` is in fact a occupied keyword which refer to `window` in main thread and `global` in a worker. So it's not really good to shadow it in the code. Some people choose to use `_this` or `THIS` or `that` or some other tricks. They're still unnecessary when you can use `bind` or  Arrow Functions, but better to redefine the exiting `self`.